components/functional/CustomScroll.tsx

```diff
@@ -0,0 +1,32 @@
+import { useEffect } from 'react';
+
+// open all external links in another tab
+export default function CustomScroll() {
+  useEffect(() => {
+    // Defining a function to handle click events on the navigation menu links
+    const handleLinkClick = (event : any) => {
+      event.preventDefault();
+      const target = event.target.getAttribute('data-href'); // Get the custom attribute
+      const element = document.querySelector(target);
+      if (element) {
+        const topOffset = element.offsetTop;
+        window.scrollTo({
+          top: topOffset - 200,
+          behavior: 'smooth',
+        });
+      }
+    };
+
+    // Select buttons with the custom attribute
+    const links = document.querySelectorAll('[data-href^="#"]');
+    links.forEach((link) => {
+      link.addEventListener('click', handleLinkClick);
+    });
+
+    return () => {
+      links.forEach((link) => {
+        link.removeEventListener('click', handleLinkClick);
+      });
+    };
+  }, []);
```

Comment:
I really liked the way you solved it!
Can i give 2 suggestions? 

1 - Despite header height can be `200` this will be hardcoded and any future changes on Header can cause some style problems, in addittion with that, our small devices header has a different height, so we can have some difficults here. 

What do you think about create some custom Hook to handle header height dynamically?

2 - I think we can handle this without adding an  `useEffect`. Dont you think that all that process of adding a click event listener than remove that click can be a bit confuse?

What about create that hook and  then return that `handleClickFunction` from that created Hook? After that we can just call our hook and then use that funcion on `onClick`